src/common/scaled-rect-buffer.c

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: GPL-2.0-only
+#define _POSIX_C_SOURCE 200809L
+#include <assert.h>
+#include <stdlib.h>
+#include <string.h>
+#include <wayland-server-core.h>
+#include <wlr/util/log.h>
+#include "buffer.h"
+#include "common/graphic-helpers.h"
+#include "common/list.h"
+#include "common/macros.h"
+#include "common/mem.h"
+#include "common/scaled-scene-buffer.h"
+#include "common/scaled-rect-buffer.h"
+
+static struct wl_list cached_buffers = WL_LIST_INIT(&cached_buffers);
+
+static void
+draw_rectangle(cairo_t *cairo, int w, int h)
```

Comment:
I think it would be better to pass the border thickness to this function in case we want to draw thicker borders in future - and so that the offset can be handled here. 

IIRC at a line-thickness of one we have to offset by 0.5. Otherwise cairo draws the color fainter across two pixels (wide).

Example: https://github.com/labwc/labwc/blob/c2928027becba26f058b544d4bd8c72895c8d7cf/src/theme.c#L1152

Reply:
Done.